packages/libs/components/src/components/plotControls/PlotGradientLegend.tsx

```diff
@@ -26,36 +21,34 @@ const legendEllipsis = (label: string, ellipsisLength: number) => {
 export default function PlotGradientLegend({
   legendMax,
   legendMin,
-  gradientColorscaleType,
-  nTicks,
+  valueToColorMapper,
+  nTicks = 5,
   showMissingness,
 }: PlotLegendGradientProps) {
   // Declare constants
   const gradientBoxHeight = 150;
   const gradientBoxWidth = 20;
   const tickLength = 4;
-  const defaultNTicks = 5;
   const tickFontSize = '0.8em';
   const legendTextSize = '1.0em';
 
-  nTicks = nTicks || defaultNTicks;
-
-  let gradientColorscale =
-    gradientColorscaleType == 'divergent'
-      ? DivergingGradientColorscale
-      : SequentialGradientColorscale;
-
-  // Create gradient stop points from the colorscale
-  const stopPoints = gradientColorscale.map((color: string, index: number) => {
-    let stopPercentage = (100 * index) / (gradientColorscale.length - 1) + '%';
-    return (
-      <stop
-        offset={stopPercentage}
-        stopColor={color}
-        key={'gradientStop' + index}
-      />
-    );
-  });
+  // Create gradient stop points from the colorscale from values [legendMin TO legendMax] at nTicks steps
+  // Do we need to use more than 5 stop points to accurately reproduce the original color gradient?
```

Comment:
My first thought was yes, five seems fine. But then i remembered the diverging colorscale - it moves through more colors so would be harder to capture with just five stop points. 
I compared with the old legend and confirmed it looks like the 5 stop points is having trouble capturing the gradient. I'd probably bump this up to 7 or 9 for diverging/converging gradients. I'll add a pic in the comments to show what i mean

Reply:
Right - thanks, yes, I thought I saw similar but didn't do the side-to-side comparison.  That is clearly not right.

We could just use 50 (100?) stop points and be done with it? This one doesn't need to be an odd number, like the text-tic marks.  There are 21 "stop points" in the original `DivergingGradientColorscale` but this component doesn't know that any more!

Reply:
I just [pushed](https://github.com/VEuPathDB/web-monorepo/pull/42/commits/cb8db5e151a37847d08fcbe95539cad98f09ec9f)  an arbitrary 50 stop points and it has improved dramatically.

![image](https://user-images.githubusercontent.com/308639/230353116-fa0288fb-d486-435b-961c-484270fd2eb1.png)